src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -419,16 +402,94 @@ def update_csv_on_market_cycle(self, slot_no: int, area: "Area") -> None:
+            area: "Area", simulation_status: str) -> None:
+        raise NotImplementedError
+
+    @classmethod
+    def _update_area_stats(cls, area: "Area", endpoint_buffer: "SimulationEndpointBuffer") -> None:
+        return
+
+    def update_csv_files(self, slot_no: int, current_time_slot: DateTime, area: "Area",
+                         scm_manager: SCMManager) -> None:
+        """Update the csv results on market cycle."""
+        if self.export_results_on_finish:
+            self._export.data_to_csv(area, current_time_slot, slot_no == 0, scm_manager)
+
+    def save_csv_results(self, area: "Area") -> None:
+        """Update the CSV results on finish, and write the CSV files."""
+        if self.export_results_on_finish:
+            log.info("Exporting simulation data.")
+            self._export.data_to_csv(area, False, None)
+            self._export.area_tree_summary_to_json(self._endpoint_buffer.area_result_dict)
+            self._export.export(power_flow=None)
+
+    def update_send_coefficient_results(
+            self, current_state: dict, progress_info: SimulationProgressInfo,
+            area: "CoefficientArea", simulation_status: str,
+            scm_manager: Optional["SCMManager"] = None) -> None:
+        """
+        Update the coefficient simulation results.
+        """
+        assert self._endpoint_buffer is not None
+
+        if self._should_send_results_to_broker:
+            self._endpoint_buffer.update_coefficient_stats(
+                area, simulation_status, progress_info, current_state,
+                False, scm_manager)
+            results = self._endpoint_buffer.prepare_results_for_publish()
+            if results is None:
+                return
+            self.kafka_connection.publish(results, current_state["simulation_id"])
+
+        elif (gsy_e.constants.RETAIN_PAST_MARKET_STRATEGIES_STATE or
+              self.export_results_on_finish):
+
+            self._endpoint_buffer.update_coefficient_stats(
+                area, current_state["sim_status"], progress_info, current_state,
+                True, scm_manager)
+            self._update_area_stats(area, self._endpoint_buffer)
+
+            if self.export_results_on_finish:
+                assert self._export is not None
+                if (progress_info.current_slot_time is not None
+                        and gsy_e.constants.RETAIN_PAST_MARKET_STRATEGIES_STATE):
+                    # for integration tests:
+                    self._export.raw_data_to_json(
+                        progress_info.current_slot_str,
+                        self._endpoint_buffer.flattened_area_core_stats_dict
+                    )
+
+                self._export.file_stats_endpoint(area, scm_manager)
+
+
+def simulation_results_manager_factory():
+    if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS.value:
+        return CoefficientSimulationResultsManager
```

Comment:
AFAIS we set the `MARKET_TYPE` variable in the setup file, which will not be evaluated until the `_init` method with `load_setup_module`. The current order is:
1. `run_simulation` with `simulation_class_factory` 
2.  `Simulation`.`__init__` with `simulation_results_manager_factory`
3.  `_init` with `load_setup_module` when we assign MARKET_TYPE within the setup file
So we probably need to introduce a new parameter for cli to include the coefficient type or use the --settings-file instead, which probably will be less work than the first option.

Reply:
I've been thinking about this a lot in the past too, that maybe adding the market type in the CLI arguments is probably the most convenient choice here. Let me quickly add it.